### PR TITLE
onepad: Maps the menu button to the analog button.

### PR DESCRIPTION
**Pull request by lukedupin**
Allows the user to toggle the analog stick with the menu button.  Fix
for Sky Odyssey.

### CODE REVIEW

File: plugins/onepad/SDL/joystick.cpp [hunk 1]
```diff
@@ -267,8 +268,9 @@ int JoystickInfo::GetInput(gamePadValues input)
 
     // Handle analog inputs which range from -32k to +32k. Range conversion is handled later in the controller
     if (IsAnalogKey(input)) {
-        int value = SDL_GameControllerGetAxis(m_controller, (SDL_GameControllerAxis)m_pad_to_sdl[input]);
-        value *= k;
+        float f_value = (float)SDL_GameControllerGetAxis(m_controller, (SDL_GameControllerAxis)m_pad_to_sdl[input]);
+        f_value *= k;
+        int value = (int)f_value;
```
Comment (turtleli) [inline plugins/onepad/SDL/joystick.cpp:273]: This change looks unnecessary.

File: plugins/onepad/wx_dialog/dialog.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ enum gui_buttons {
 #define BUTTONS_LENGHT 32 // numbers of buttons on the gamepad
 #define UPDATE_TIME 5
 #define DEFAULT_WIDTH 1000
-#define DEFAULT_HEIGHT 740
+#define DEFAULT_HEIGHT 820
```
Comment (turtleli) [inline plugins/onepad/wx_dialog/dialog.h:64]: Wouldn't this negatively affect people with smaller screens?

**Comment by lightningterror** [comment]
@lukedupin Status?

**Comment by lukedupin** [comment]
@lightningterror  I don't have a new status update. When the code was pushed, everything was tested and working. This is still the build I use on my machine. My Apologizes but I don't have time to check the rebase conflict that is listed above.

**Comment by lightningterror** [comment]
There are some review comments that turtleli left too btw.

File: plugins/onepad/SDL/joystick.cpp (hunk 1 above)
Comment (lukedupin) [reply plugins/onepad/SDL/joystick.cpp:273]: Functionally it is the same. Old habit of being explicit.

File: plugins/onepad/wx_dialog/dialog.h (hunk 2 above)
Comment (lukedupin) [reply plugins/onepad/wx_dialog/dialog.h:64]: https://www.w3schools.com/browsers/browsers_display.asp According to this ~30% of screens will be negatively impacted. Its probably possible to get the resolution and then set the value dynamically?

**Comment by CristianPi** [comment]
Amazing, thanks! 
Also useful with Klonoa 2 on linux (as LilyPad does not work).

**Comment by lightningterror** [comment]
Needs to be rebased as onepad was merged in to core.

**Comment by GovanifY** [comment]
Remade most of the PR as PAD got merged into core, took care of review comments too, this should be good to go!

**Comment by RedDevilus** [comment]
@lukedupin Could you rebase this PR so we can merge it as a development build?

**Comment by lukedupin** [comment]
@RedDevilus I'm pretty far out of the loop on this code. When I submitted the pull request it was ready. I really don't know whats changed since then and I don't have time to dive back into this.

**Comment by lightningterror** [comment]
I will close this then since it's not being worked on, if you wish to continue we can reopen the pr.

**Pull request closed without merging**